jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -698,6 +741,40 @@ def apply_RGB_presets(self):
     def vue_apply_RGB_presets(self, data):
         self.apply_RGB_presets()
 
+    @observe('viewer_selected', 'viewer_x_max_value', 'viewer_x_min_value',
+             'viewer_y_max_value', 'viewer_y_min_value')
+    def _update_viewer_bound_steps(self, msg={}):
+        if not hasattr(self, 'viewer'):  # pragma: no cover
+            # plugin hasn't been fully initialized yet
+            return
+
+        if not self.viewer.selected:  # pragma: no cover
+            # nothing selected yet
+            return
+
+        viewer = self.viewer.selected_obj[0] if self.viewer_multiselect else self.viewer.selected_obj # noqa
+        if not isinstance(viewer.state, ImageViewerState):
+            # We round these values to show, e.g., 7.15 instead of 7.1499999
+            if hasattr(viewer.state, "x_max") and viewer.state.x_max is not None:
+                bound_step = (viewer.state.x_max - viewer.state.x_min) / 100.
+                decimals = -int(np.log10(abs(bound_step))) + 1 if bound_step != 0 else 6
+                self.viewer_x_bound_step = np.round(bound_step, decimals=decimals)
+                self.viewer_x_max_value = np.round(self.viewer_x_max_value, decimals=decimals)
+                self.viewer_x_min_value = np.round(self.viewer_x_min_value, decimals=decimals)
+            if hasattr(viewer.state, "y_max") and viewer.state.y_max is not None:
+                bound_step = (viewer.state.y_max - viewer.state.y_min) / 100.
+                decimals = -int(np.log10(abs(bound_step))) + 1 if bound_step != 0 else 6
+                self.viewer_y_bound_step = np.round(bound_step, decimals=decimals)
+                self.viewer_y_max_value = np.round(self.viewer_y_max_value, decimals=decimals)
+                self.viewer_y_min_value = np.round(self.viewer_y_min_value, decimals=decimals)
+
+    def vue_reset_viewer_bounds(self, _):
+        # This button is currently only exposed if only the spectrum viewer is selected
+        viewer = self.app.get_viewer('spectrum-viewer')

```

Comment:
can we use the selected viewer reference directly instead of hardcoding for spectrum-viewer?